src/seedu/addressbook/logic/Logic.java

```diff
@@ -66,15 +68,30 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
 
     /**
      * Parses the user command, executes it, and returns the result.
+     * Record this user command.
      * @throws Exception if there was any problem during command execution.
      */
     public CommandResult execute(String userCommandText) throws Exception {
+        userCommandText = checkLastCommand(userCommandText);
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
         recordResult(result);
         return result;
     }
 
+    /**
+     * Check if the command is "last".
+     * If yes then append the lastCommand for later execution.
+     */
+    private String checkLastCommand(String userCommandText) throws Exception {
+        if( userCommandText.substring(0, Math.min(userCommandText.length(), 4)).equals("last")) {
```

Comment:
Avoid the magical string usage like "last". You have already defined it as a constant COMMAND_WORD. So please make use of it